mlflow/pytorch/_lightning_autolog.py

```diff
@@ -438,15 +359,25 @@ def on_train_batch_end(
         batch,
         batch_idx,
     ) -> None:
+        self.trainer = trainer
```

Comment:
I'm a little concerned about this line - we are setting a class attribute in `on_train_batch_end`, which is called multiple times. Do we need this line?

Reply:
I can make it only set it at the first time.

The purpose of setting this attribute is for usage in `save_checkpoint`, otherwise we need to figure out a way to pass `trainer` object to `save_checkpoint` function. Note `save_checkpoint` is called in base class but its implementation is in sub-class.

Reply:
can we set it in the constructor or in `on_train_start ` hook?

Reply:
Moved it into constructor